File: Practica de 121/POO/1-persona/Persona.java
//Crea una clase Persona con nombre, edad y ciudad
public class Persona {
    private String nombre;
    private int edad;
    private String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    // a) Método para mostrar saludo
    public String saludar() {
        return "Hola, soy " + this.nombre + " de " + this.ciudad;
    }

    // c) Método para verificar mayoría de edad
    public boolean esMayorDeEdad() {
        return this.edad >= 18;
    }

    // b) Main para crear tres personas y mostrar su saludo
    public static void main(String[] args) {
        Persona persona1 = new Persona("Juan", 25, "Madrid");
        Persona persona2 = new Persona("María", 17, "Barcelona");
        Persona persona3 = new Persona("Carlos", 30, "Valencia");

        System.out.println(persona1.saludar());
        System.out.println(persona2.saludar());
        System.out.println(persona3.saludar());

        System.out.println(persona1.nombre + " es mayor de edad: " + persona1.esMayorDeEdad());
        System.out.println(persona2.nombre + " es mayor de edad: " + persona2.esMayorDeEdad());
        System.out.println(persona3.nombre + " es mayor de edad: " + persona3.esMayorDeEdad());
    }
}

File: Practica de 121/POO/1-persona/Persona.py
#Crea una clase Persona con nombre, edad y ciudad
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad
    
    # a) Método para mostrar saludo
    def saludar(self):
        return f"Hola, soy {self.nombre} de {self.ciudad}"
    
    # c) Método para verificar mayoría de edad
    def es_mayor_de_edad(self):
        return self.edad >= 18

# b) Crear tres personas y mostrar su saludo
persona1 = Persona("Juan", 25, "Madrid")
persona2 = Persona("María", 17, "Barcelona")
persona3 = Persona("Carlos", 30, "Valencia")

print(persona1.saludar()) 
print(persona2.saludar()) 
print(persona3.saludar())  

print(f"{persona1.nombre} es mayor de edad: {persona1.es_mayor_de_edad()}")  
print(f"{persona2.nombre} es mayor de edad: {persona2.es_mayor_de_edad()}")  
print(f"{persona3.nombre} es mayor de edad: {persona3.es_mayor_de_edad()}")  